crates/zed/src/languages/sql/highlights.scm

```diff
@@ -0,0 +1,114 @@
+(string) @string
```

Comment:
I've tried the highlights for `zed/crates/collab/migrations/` files and found that we seem to highlight keywords inside field names which looks wrong:

![image](https://github.com/zed-industries/zed/assets/2690773/18ff1569-029a-4e17-b0b8-d1319159451f)
![image](https://github.com/zed-industries/zed/assets/2690773/cc443722-e1e5-44a7-ab15-ae5bc1bbdb3d)

And generally, consistently highlighted fields look quite good:
![image](https://github.com/zed-industries/zed/assets/2690773/eb814c94-c480-4fdb-9e0d-b6cc5a2e536b)
I wonder, if we could do that + maybe table names? Can we do the type declarations too?


And when I checked it on 
<details>
<summary>some prompt-generated example</summary>

```sql
CREATE TABLE sales (
    id SERIAL PRIMARY KEY,
    employee VARCHAR(50),
    sale_date DATE,
    amount NUMERIC
);

INSERT INTO sales (employee, sale_date, amount) VALUES
('John', '2023-01-01', 100.00),
('John', '2023-01-02', 120.00),
('Jane', '2023-01-02', 130.00),
('John', '2023-01-03', 110.00),
('Jane', '2023-01-03', 140.00),
('Mike', '2023-01-03', 150.00);

-- Query to aggregate total sales per employee and rank them
WITH total_sales_ranked AS (
    SELECT
        employee,
        SUM(amount) AS total_sales,
        RANK() OVER (ORDER BY SUM(amount) DESC) AS sales_rank
    FROM
        sales
    GROUP BY
        employee
),
-- Query to get detailed sales records
detailed_sales AS (
    SELECT
        id,
        employee,
        sale_date,
        amount,
        'Detailed Record' AS record_type
    FROM
        sales
),
-- Query to get aggregated sales records with rank
aggregated_sales AS (
    SELECT
        NULL AS id,  -- Since aggregated result doesn't have a unique ID
        employee,
        NULL AS sale_date,  -- No specific date for aggregated result
        total_sales AS amount,
        'Aggregated Total' AS record_type,
        sales_rank
    FROM
        total_sales_ranked
)

-- Combine detailed and aggregated sales using UNION ALL
SELECT
    *
FROM
    detailed_sales
UNION ALL
SELECT
    id,
    employee,
    sale_date,
    amount,
    record_type
FROM
    aggregated_sales
ORDER BY
    employee, record_type DESC, id;

SELECT
  id,
  employee,
  sale_date,
  amount,
  ROW_NUMBER() OVER (PARTITION BY employee ORDER BY sale_date) as sale_row_number
FROM
  sales;
```

</details>

something complex like window functions is not highlighted well (but is that even feasible to do now?)

![image](https://github.com/zed-industries/zed/assets/2690773/0a02ca82-0376-4bde-bcf0-71292eefc69d)

Reply:
Humm, thanks for the great review, I'm going to try address those issues ASAP :) and is a try because this is my first contact with tree-sitter :)

Reply:
I'm afraid I cannot help you here too, but I've seen that other people copied parts of other open-source editors' tree-sitter highlight rules, or just got help from the community in the issue — the SQL one is quite popular and somebody might be able to provide help.

Reply:
Yep, I think that is good way to understand and contribute to the project, by understanding others implementations.

Reply:
<img width="602" alt="image" src="https://github.com/zed-industries/zed/assets/582516/f2c57522-66f9-41a5-84bb-11c8e315959d">

Looks like that the tree-sitter-sql under the hood doesn't have support for many aspects of the SQL, for instance in this case if we add `COLUMN` after the `ADD` it works.

Actually I'm not sure if using the `highlights.scm` in the correct way (copied from the https://github.com/m-novikov/tree-sitter-sql).

Actually it will be awesome if the original author can shed some light here :D

Reply:
Ok today I start to learn the tree-sitter for good LOL, and I'm able to fork and started to modify the grammar to be compatible with postgres